rustls/src/client/tls13.rs

```diff
@@ -147,6 +147,12 @@ pub(super) fn handle_server_hello(
         KeySchedulePreHandshake::new(suite)
     };
 
+    cx.common.kx_state = match cx.common.kx_state {
```

Comment:
maybe this line and the one below could be just `cx.common.kx_state.complete()`?

(alongside a `fn complete(&mut self) { ... }` on `KxState`)

could take it or leave it

Reply:
I like :+1: I ended up replacing the `unreachable!` and went with:

```rust
    pub(crate) fn complete(&mut self) {
        debug_assert!(matches!(self, Self::Start(_)));
        if let Self::Start(group) = self {
            *self = Self::Complete(*group);
        }
    }
```

I think that's the right balance - if this assert invariant didn't hold in a prod build it feels better to have `negotiated_kx_group()` return `None` as opposed to panic'ing in `KxState::complete`.